addons/mail/static/src/core/common/message_model.js

```diff
@@ -399,7 +399,7 @@ export class Message extends Record {
             !this.is_transient &&
                 !this.isPending &&
                 this.thread &&
-                this.store.self.type === "partner" &&
+                this.store.self_partner &&
```

Comment:
I think this check is useless, we wouldn't have a `main_user_id` on a guest.
Even store.self at the next line could be replaced by `self_partner`.